pkg/api/validation/validation.go

```diff
@@ -397,6 +401,20 @@ func validateISCSIVolumeSource(iscsi *api.ISCSIVolumeSource) errs.ValidationErro
 	return allErrs
 }
 
+func validateFCVolumeSource(fc *api.FCVolumeSource) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if len(fc.TargetWWNs) < 1 {
+		allErrs = append(allErrs, errs.NewFieldRequired("targetWWNs"))
+	}
+	if fc.FSType == "" {
+		allErrs = append(allErrs, errs.NewFieldRequired("fsType"))
+	}
+	if fc.Lun < 0 || fc.Lun > 255 {
```

Comment:
We try to avoid having 0 mean "not specified" - if 0 is not a valid value
but the field is optional, use a pointer to an int.

On Fri, Aug 28, 2015 at 7:30 PM, Mark Turansky <notifications@github.com>
wrote:

> In pkg/api/validation/validation.go
> <https://github.com/kubernetes/kubernetes/pull/12599#discussion_r38250767>
> :
>
> > @@ -397,6 +401,20 @@ func validateISCSIVolumeSource(iscsi *api.ISCSIVolumeSource) errs.ValidationErro
> >  	return allErrs
> >  }
> >
> > +func validateFCVolumeSource(fc *api.FCVolumeSource) errs.ValidationErrorList {
> > +	allErrs := errs.ValidationErrorList{}
> > +	if len(fc.TargetWWNs) < 1 {
> > +		allErrs = append(allErrs, errs.NewFieldRequired("targetWWNs"))
> > +	}
> > +	if fc.FSType == "" {
> > +		allErrs = append(allErrs, errs.NewFieldRequired("fsType"))
> > +	}
> > +	if fc.Lun < 0 || fc.Lun > 255 {
>
> Alternatively, change it from Required to Optional in types.go and make
> clear that 0 is the default value.
>
> As is, you did not make it a required field. The default value would work,
> so the API for this object is incorrect.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/12599/files#r38250767>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift